src/Notifier.ts

```diff
@@ -435,7 +435,14 @@ export const Notifier = {
         if (actions?.notify) {
             this._performCustomEventHandling(ev);
 
-            if (SdkContextClass.instance.roomViewStore.getRoomId() === room.roomId &&
+            const store = SdkContextClass.instance.roomViewStore;
+            const isViewingRoom = store.getRoomId() === room.roomId;
+            const threadId: string | undefined = ev.getThread()?.id;
+            const isViewingThread = store.getThreadId() === threadId;
+
+            const isViewingEventTimeline = (isViewingRoom && !threadId) || (threadId && isViewingThread);
```

Comment:
Might be more readable as

```suggestion
            const isViewingEventTimeline = isViewingRoom && (!threadId || isViewingThread);
```